homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,229 @@
+                        ],
+                    },
+                )
+        try:
+            main_user = self._get_lastfm_user(self.data[CONF_MAIN_USER])
+            friends: Sequence[SelectOptionDict] = [
+                {"value": str(friend.name), "label": str(friend.get_name(True))}
+                for friend in main_user.get_friends()
+            ]
+        except WSError:
+            friends = []
+        return self.async_show_form(
+            step_id="friends",
+            errors=errors,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERS, default=valid_users): SelectSelector(
+                        SelectSelectorConfig(
+                            options=friends, custom_value=True, multiple=True
+                        )
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import config from yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        return self.async_create_entry(
+            title="LastFM",
+            data={
+                CONF_API_KEY: import_config[CONF_API_KEY],
+                CONF_MAIN_USER: None,
+                CONF_USERS: import_config[CONF_USERS],
+            },
+        )
+
+
+class LastFmOptionsFlowHandler(OptionsFlow, LastFmFlowHandler):
+    """LastFm Options flow handler."""
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize LastFM Options flow."""
+        super().__init__(dict(entry.data))
+        self.entry = entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize form."""
+        valid_users = self.data[CONF_USERS]
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for user in user_input[CONF_USERS]:
+                lastfm_user = self._get_lastfm_user(user)
+                lastfm_errors = self._validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(user)
+            if not errors:
+                await self.hass.config_entries.async_unload(self.entry.entry_id)
+                for username in self.data[CONF_USERS]:
+                    if username not in user_input[CONF_USERS] and (
+                        entity_id := er.async_get(self.hass).async_get_entity_id(
+                            Platform.SENSOR, DOMAIN, f"sensor.lastfm_{username}"
+                        )
+                    ):
+                        er.async_get(self.hass).async_remove(entity_id)
+                await self.hass.config_entries.async_reload(self.entry.entry_id)
```

Comment:
Make a [listener](https://github.com/home-assistant/core/blob/dev/homeassistant/components/yale_smart_alarm/__init__.py#L30) in __init__ instead.

Also adding/removing entities should probably be done in `sensor` and not here

Reply:
Do you have an example?

Reply:
Don't think we should do that. After reload they would become obsolete so the user should remove them.

Reply:
Whoops, didn't see the link in your first comment, added it now

Reply:
Is there a way I can make that testable?